hibernate-core/src/main/java/org/hibernate/hql/internal/ast/HqlSqlWalker.java

```diff
@@ -1089,9 +1089,9 @@ protected void beforeSelectClause() throws SemanticException {
 	@Override
 	protected AST generatePositionalParameter(AST delimiterNode, AST numberNode) throws SemanticException {
 		// todo : we check this multiple times
-		if ( namedParameters != null ) {
+		if (  getSessionFactoryHelper().isStrictJPAQLComplianceEnabled() && namedParameters != null ) {
 			throw new SemanticException(
-					"Cannot define positional and named parameterSpecs : " + queryTranslatorImpl.getQueryString()
+					"Cannot mix positional and named parameters : " + queryTranslatorImpl.getQueryString()
```

Comment:
I know it was preexisting but can you remove the space before the colon? It should be `and named parameters: " +`